doc/stdenv/meta.chapter.md

```diff
@@ -134,7 +134,27 @@ Attribute Set `lib.platforms` defines [various common lists](https://github.com/
 This attribute is special in that it is not actually under the `meta` attribute set but rather under the `passthru` attribute set. This is due to how `meta` attributes work, and the fact that they are supposed to contain only metadata, not derivations.
 :::
 
-An attribute set with as values tests. A test is a derivation, which builds successfully when the test passes, and fails to build otherwise. A derivation that is a test needs to have `meta.timeout` defined.
+An attribute set with tests as values. A test is a derivation that builds when the test passes and fails to build otherwise.
+
+#### Regular tests
+
+Tests that are part of the source package are often executed in the `installCheckPhase`. Prefer `passthru.tests` for tests that are introduced in nixpkgs because `installCheckPhase` adds overhead to each build and we can run `passthru.tests` independently.
```

Comment:
Sorry for reacting a bit late, life is busy ;). I think it would be good to have two motivations here: (1) too much overhead, and (2) where one really wants to test the derivation as a dependency. The problem with `installCheckPhase` is that it still runs within the environment of the derivation, so it may not catch missing dependencies, missing files, etc.

To give an example, suppose that a C library should have another library in its `propagatedBuildInputs`, but it is in the `buildInputs`. You wouldn't detect this with `installCheckPhase`, since it gets the dependency's environment through `buildInputs`. However, a passthru test that builds against the C library would catch the missing propagated build input, since it is not built in the original environment of the derivation.

It is true that downstream users of the C library could also break due to such an error. However, downstream users break for all kinds of reasons, or the maintainer may have added the dependency through `buildInputs`, though it should actually have been propagated.